mono/tests/roslyn-bug-19038.cs

```diff
@@ -0,0 +1,35 @@
+using System;
+using System.Reflection;
+
+unsafe public class C {
+    public int Value;
+
+    static void Main()
+    {
+        C a = new C { Value = 12 };
+        FieldInfo info = typeof(C).GetField("Value");
+        TypedReference reference = __makeref(a);
+
+        if (!(reference is TypedReference reference0)) 
+            throw new Exception("TypedReference");
+
+        info.SetValueDirect(reference0, 34);
+
+        Console.WriteLine($"a.Value = {a.Value}");
+        if (a.Value != 34)
+            throw new Exception("SetValueDirect");
+
+        int z = 56;
+        if (CopyRefInt(ref z) != 56) 
```

Comment:
Is this testing the same bug? Does CopyRefInt use a typed reference under the hood somehow?

Reply:
I don't know anything about the test itself, it just happened to use SetValueIndirect. I think they're just testing the overall code generation, which is why it has multiple throws.

Reply:
I think it's worth checking the IL if this compiles to `SetValueIndirect` and if it doesn't, leave it out of this test.